pandas/core/internals.py

```diff
@@ -2734,6 +2797,13 @@ class SparseBlock(NonConsolidatableMixIn, Block):
     _holder = SparseArray
     _concatenator = staticmethod(_concat._concat_sparse)
 
+    def __init__(self, values, placement, ndim=None):
+        # Ensure that we have the underlying SparseArray here...
```

Comment:
so this should be on ExtensionArray. I know that Sparse is not there yet, but the idea is that you should call a method to do this coercion, IOW so each sub-class doesn't have to repeat it.

Reply:
Agreed. We should be able to remove the `__init__` for `DatetimeBlock`, `CategoricalBlock`, and `SparseBlock`.